pkg/loki/loki_test.go

```diff
@@ -90,10 +91,39 @@ func TestLoki_isModuleEnabled(t1 *testing.T) {
 	}
 }
 
+func getRandomPorts(n int) []int {
```

Comment:
big fan of using random ports in test! I do wonder if this is overly complicated though with the loop and multiple ports. why this approach vs a function that returns a random, free port that gets called multiple times, ie:

```
httpPort := getRandomPort()
grpcPort := getRandomPort()
```

Reply:
Hmm just for being extra safe. Since at the end of `getRandomPort` the listener is freed, there's an (absurdly small) chance of the first and second call returning the same port.